tensornetwork/tests/network_operations_test.py

```diff
@@ -217,3 +217,36 @@ def test_reachable(backend):
   nodes = [tn.Node(np.random.rand(2, 2, 2), backend=backend) for _ in range(10)]
   _ = [nodes[n][0] ^ nodes[n + 1][1] for n in range(len(nodes) - 1)]
   assert set(nodes) == tn.reachable(nodes[0])
+
+
+def test_reachable_disconnected_1(backend):
+  nodes = [tn.Node(np.random.rand(2, 2, 2), backend=backend) for _ in range(4)]
+  nodes[0][1] ^ nodes[1][0]
+  nodes[2][1] ^ nodes[3][0]
+  assert set(tn.reachable([nodes[0], nodes[2]])) == set(nodes)
+
+  assert set(tn.reachable([nodes[0]])) == {nodes[0], nodes[1]}
+  assert set(tn.reachable([nodes[1]])) == {nodes[0], nodes[1]}
+  assert set(tn.reachable([nodes[0], nodes[1]])) == {nodes[0], nodes[1]}
+
+  assert set(tn.reachable([nodes[2]])) == {nodes[2], nodes[3]}
+  assert set(tn.reachable([nodes[3]])) == {nodes[2], nodes[3]}
+  assert set(tn.reachable([nodes[2], nodes[3]])) == {nodes[2], nodes[3]}
+
+  assert set(tn.reachable([nodes[0], nodes[1],
```

Comment:
Why does reachable with `nodes[0]` not also return `nodes[0]`?

Reply:
Based on this failing test, it seems like the answer is "It does"

```
 assert set(tn.reachable([nodes[0], nodes[1],
                               nodes[2]])) == {nodes[2], nodes[3]}
E     assert {<tensornetwo...7f7120f1c4a8>} == {<tensornetwor...7f7120f1c4a8>}
E       Extra items in the left set:
E       <tensornetwork.network_components.Node object at 0x7f7120f03cc0>
E       <tensornetwork.network_components.Node object at 0x7f7120f03668>
E       Use -v to get the full diff
```

Reply:
thanks, sorry hadn't fixd them yet